dot/parachain/candidate-validation/candidate_validation_test.go

```diff
@@ -677,3 +669,215 @@ func Test_performBasicChecks(t *testing.T) {
 		})
 	}
 }
+
+func TestCandidateValidation_validateFromChainState(t *testing.T) {
+	t.Parallel()
+	candidateReceipt, validationCode := createTestCandidateReceiptAndValidationCode(t)
+	candidateReceipt2 := candidateReceipt
+	candidateReceipt2.Descriptor.PovHash = common.MustHexToHash(
+		"0x1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef")
+	candidateReceipt2.Descriptor.ParaID = 2
+	candidateReceipt3 := candidateReceipt
+	candidateReceipt3.Descriptor.ParaID = 3
+	candidateReceipt4 := candidateReceipt
+	candidateReceipt4.Descriptor.ParaID = 4
+	candidateReceipt4.Descriptor.ValidationCodeHash = parachaintypes.ValidationCodeHash(common.MustHexToHash(
+		"0x1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef"))
+	candidateReceipt5 := candidateReceipt
+	candidateReceipt5.Descriptor.ParaID = 5
```

Comment:
Do a bit better indentation. Kinda like this

```
	t.Parallel()

	candidateReceipt, validationCode := createTestCandidateReceiptAndValidationCode(t)
	candidateReceipt2 := candidateReceipt
	candidateReceipt2.Descriptor.PovHash = common.MustHexToHash(
		"0x1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef")
	candidateReceipt2.Descriptor.ParaID = 2

	candidateReceipt3 := candidateReceipt
	candidateReceipt3.Descriptor.ParaID = 3

	candidateReceipt4 := candidateReceipt
	candidateReceipt4.Descriptor.ParaID = 4
	candidateReceipt4.Descriptor.ValidationCodeHash = parachaintypes.ValidationCodeHash(common.MustHexToHash(
		"0x1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef"))

	candidateReceipt5 := candidateReceipt
	candidateReceipt5.Descriptor.ParaID = 5
```

Reply:
Added line breaks for clearer grouping.